src/map/status.hpp

```diff
@@ -2465,6 +2491,97 @@ enum e_status_bonus {
 	STATUS_BONUS_RATE = 1,
 };
 
+///Enum of Status Change Flags [Cydh]
+enum e_status_change_flag : uint16 {
+	SCF_NONE = 0,
+	SCF_BLEFFECT,
+	SCF_DISPLAY_PC,
+	SCF_NO_CLEARBUFF,
+	SCF_NO_REM_ONDEAD,
+	SCF_NO_DISPELL,
+	SCF_NO_CLEARANCE,
+	SCF_NO_BANISHING_BUSTER,
+	SCF_NO_SAVE,
+	SCF_REM_ON_DAMAGED,
+	SCF_REM_ON_REFRESH,
+	SCF_REM_ON_LUXANIMA,
+	SCF_STOP_ATTACKING,
+	SCF_STOP_CASTING,
+	SCF_STOP_WALKING,
+	SCF_BOSS_RESIST,
+	SCF_MVP_RESIST,
+	SCF_SET_STAND,
+	SCF_FAILED_MADO,
+	SCF_DEBUFF,
+	SCF_REM_ON_CHANGEMAP,
+	SCF_REM_ON_MAPWARP,
+	SCF_REM_CHEM_PROTECT,
+	SCF_OVERLAP_FAIL,
+	SCF_OPT_CHANGEOPTION,
+	SCF_OPT_CHANGELOOK,
+	SCF_TRIGGER_ONTOUCH_,
+	SCF_DISPLAY_NPC,
+	SCF_REQUIRE_WEAPON,
+	SCF_REQUIRE_SHIELD,
+	SCF_MAX
+};
+
+/// Struct of SC configs [Cydh]
+struct s_status_change_db {
+	sc_type type;				///< SC_
+	efst_type icon;				///< EFST_
+	uint32 state;				///< SCS_
+	uint32 calc_flag;			///< SCB_ flags
+	uint16 opt1;				///< OPT1_
+	uint16 opt2;				///< OPT2_
+	uint32 opt3;				///< OPT3_
+	uint32 look;				///, OPTION_ Changelook
+	std::bitset<SCF_MAX> flag;	///< SCF_ Flags, enum e_status_change_flag
+	bool display;				///< Display status effect/icon (for certain state)
+	uint16 skill_id;			///< Associated skill for (addeff) duration lookups
+	std::vector<sc_type> end;	///< List of SC that will be ended when this SC is activated
+	std::vector<sc_type> fail;	///< List of SC that causing this SC cannot be activated
+	bool end_return;			///< After SC ends the SC from end list, it does nothing
+	script_code *script;		///< Executes this script when SC starts
+	t_tick min_duration;		///< Minimum duration effect (after all status reduction)
+	uint16 min_rate;			///< Minimum rate to be applied (after all status reduction)
+
+	~s_status_change_db()
+	{
+		if (script)
+			script_free_code(script);
+	}
+};
+
+class StatusDatabase : public TypesafeYamlDatabase<int32, s_status_change_db> {
```

Comment:
This needs to use the cached version! A lot of calls will be done, for example getIcon